apis/python/src/tiledbsoma/managed_query.cc

```diff
@@ -133,6 +136,36 @@ void load_managed_query(py::module& m) {
                 }
             })
 
+        .def(
+            "next",
+            [](ManagedQuery& mq) -> std::optional<py::object> {
+                // Release python GIL before reading data
+                py::gil_scoped_release release;
+
+                mq.setup_read();
+
+                if (mq.is_empty_query() && mq.is_first_read()) {
```

Comment:
I think the code would be easier to understand if this code block was removed and empty queries went through the same code as all other queries.

All this would require is that `submit_read` do a check for an empty query, and return immediately if it is empty (i.e., move this check into submit_read). 

As part of that, you could remove the `query_submitted_ = true;` from the results method, which strikes me as an odd place to set this flag.